modal/cli/container.py

```diff
@@ -122,33 +170,50 @@ def try_write():
         return future
 
     async def _get_output():
-        nonlocal last_entry_id, completed
+        nonlocal last_batch_index, exit_status, connected
 
         req = api_pb2.ContainerExecGetOutputRequest(
             exec_id=exec_id,
             timeout=55,
-            last_entry_id=last_entry_id,
+            last_batch_index=last_batch_index,
         )
         async for batch in unary_stream(client.stub.ContainerExecGetOutput, req):
             for message in batch.items:
                 assert message.file_descriptor in [1, 2]
 
                 await write_data(message.file_descriptor, str.encode(message.message))
 
-            if batch.eof:
-                completed = True
+            if not connected:
+                connected = True
+                on_connect()
+
+            if batch.HasField("exit_code"):
+                exit_status = batch.exit_code
                 break
 
-            if batch.entry_id:
-                last_entry_id = batch.entry_id
+            if batch.batch_index:
+                last_batch_index = batch.batch_index
```

Comment:
todo: I don't think this will ever not be set, so the if statement seems unnecessary.